homeassistant/components/sensor/fritzbox.py

```diff
@@ -0,0 +1,77 @@
+"""
+Support for AVM Fritz!Box smarthome temperature sensor only devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/sensor.fritzbox/
+"""
+import logging
+
+import requests
+
+from homeassistant.components.fritzbox import DOMAIN as FRITZBOX_DOMAIN
+from homeassistant.components.fritzbox import (
+    ATTR_STATE_DEVICE_LOCKED, ATTR_STATE_LOCKED)
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELSIUS
+
+DEPENDENCIES = ['fritzbox']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Fritzbox smarthome sensor platform."""
+    _LOGGER.debug("Initializing fritzbox temperature sensors")
+    devices = []
+    fritz_list = hass.data[FRITZBOX_DOMAIN]
+
+    for fritz in fritz_list:
+        device_list = fritz.get_devices()
+        for device in device_list:
+            if (device.has_temperature_sensor
+                    and not device.has_switch
+                    and not device.has_thermostat):
+                devices.append(FritzBoxTempSensor(device, fritz))
+
+    add_entities(devices)
+
+
+class FritzBoxTempSensor(Entity):
+    """The entity class for Fritzbox temperature sensors."""
+
+    def __init__(self, device, fritz):
+        """Initialize the switch."""
+        self._device = device
+        self._fritz = fritz
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._device.name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    def update(self):
+        """Get latest data and states from the device."""
+        try:
+            self._device.update()
+        except requests.exceptions.HTTPError as ex:
```

Comment:
Just a side note (and more related to the backend lib), but its developer should consider wrapping such exceptions related to internals of the library under a `FritzException` to make it easier to catch them.